tests/unit/steps/test_base_step.py

```diff
@@ -740,100 +602,115 @@ def my_step() -> None:
         step_instance()
 
 
+@step
+def environment_test_step_1() -> None:
+    assert Environment().step_is_running is True
+
+
 def test_step_sets_global_execution_status_on_environment(one_step_pipeline):
     """Tests that the `Environment.step_is_running` value is set to
     True during step execution."""
-
-    @step
-    def my_step() -> None:
-        assert Environment().step_is_running is True
-
     assert Environment().step_is_running is False
-    one_step_pipeline(my_step()).run()
+    one_step_pipeline(environment_test_step_1()).run(unlisted=True)
     assert Environment().step_is_running is False
 
 
+@step
+def environment_test_step_2() -> None:
+    raise RuntimeError()
+
+
 def test_step_resets_global_execution_status_even_if_the_step_crashes(
     one_step_pipeline,
 ):
     """Tests that the `Environment.step_is_running` value is set to
     False after step execution even if the step crashes."""
 
-    @step
-    def my_step() -> None:
-        raise RuntimeError()
-
     assert Environment().step_is_running is False
     with pytest.raises(RuntimeError):
-        one_step_pipeline(my_step()).run()
+        one_step_pipeline(environment_test_step_2()).run(unlisted=True)
     assert Environment().step_is_running is False
 
 
+@step
+def wrong_int_output_step_1() -> int:
+    return "not_an_int"
+
+
+@step
+def wrong_int_output_step_2() -> Output(some_output_name=int):
+    return "not_an_int"
+
+
+@step
+def wrong_int_output_step_3() -> Output(out1=int, out2=int):
+    return 1, "not_an_int"
+
+
 def test_returning_an_object_of_the_wrong_type_raises_an_error(
     one_step_pipeline,
 ):
     """Tests that returning an object of a type that wasn't specified (either
     directly or as part of the `Output` tuple annotation) raises an error."""
 
-    @step
-    def some_step_1() -> int:
-        return "not_an_int"
-
-    @step
-    def some_step_2() -> Output(some_output_name=int):
-        return "not_an_int"
-
-    @step
-    def some_step_3() -> Output(out1=int, out2=int):
-        return 1, "not_an_int"
-
-    for step_function in [some_step_1, some_step_2, some_step_3]:
+    for step_function in [
+        wrong_int_output_step_1,
+        wrong_int_output_step_2,
+        wrong_int_output_step_3,
+    ]:
```

Comment:
could also be done with @pytest.mark.parametrize("step_function", [wrong_int_output_step_1, wrong_int_output_step_2, wrong_int_output_step_3]) - That way these show up as three separate tests and this test becomes more obviously extensible to more failing scenarios